_source/_posts/2021-11-02-k8s-to-the-cloud-aws.md

```diff
@@ -0,0 +1,391 @@
+
+{% img blog/k8s-to-the-cloud-aws/2.png alt:"Local sign-in page for Simple Node.js application" width:"900" %}{: .center-image }
+
+### Pushing the docker image to the Elastic Container Registry (ECR)
+
+In the last section, you built the docker image for your application. Now you can proceed to create a registry within the Elastic Container Registry, and push the newly created docker image into it.
+
+Execute the `ecr-public` command below to create a public repository within the Elastic Container Registry. The tag flag `--tags` will also attach a tag with a key of environment and value of development to the repository. This command works for the us-east-1 region, but does not work in all AWS regions.
+
+```bash
+aws ecr-public create-repository --repository-name okta-k8-ecr-repo --tags Key=environment,Value=development --region us-east-1
+```
+
+The command executed above will return a JSON response by default to describe the created repository. Take note of the `repositoryUri` contained in the JSON response, as you will attach a tag containing the `repositoryUri` to the docker image.
+
+{% img blog/k8s-to-the-cloud-aws/3.png alt:"Creating a repository within the Elastic Container Registry" width:"900" %}{: .center-image }
+
+Next, execute the docker tag command below to tag the docker image built previously.
+
+**Note:** Replace the <REPOSITORY_URI> placeholder within the template section in the code block below with the REPOSITORY_URI of the docker image you pushed to the ECR in the format of `<REPOSITORY_URI>:latest` .
+
+```bash
+docker tag okta-k8-app:latest <REPOSITORY_URI>:latest
+```
+
+Execute the command below to authenticate your computer's docker client with your ECR repo through an authentication token.
+
+```bash
+aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
+```
+
+Next, execute the command below to push the tagged docker image to the ECR repository you created.
+
+```bash
+docker push <REPOSITORY_URI>
+```
+
+{% img blog/k8s-to-the-cloud-aws/4.png alt:"Pushing the app image to the ECR repository" width:"900" %}{: .center-image }
+
+### Creating an EKS cluster
+
+So far, you have pushed a docker image to the Elastic Container Registry. Let's proceed to create a cluster in EKS that will use the docker image you previously pushed.
+
+Use the eksctl CLI tool to create a cluster within EKS. eksctl is a third-party CLI tool written in Go that simplifies the management of EKS clusters by leveraging AWS CloudFormation to manage your cluster and other dependent objects such as VPCs, and subnets.
+
+When creating a cluster, eksctl requires a very minimal configuration. It will attempt to create a cluster using its default values. However, you can specify the cluster configurations as command line arguments to the eksctl command, or more conveniently, specify a YAML file containing your cluster configurations.
```

Comment:
```suggestion
When creating a cluster, `eksctl` requires a very minimal configuration. It will attempt to create a cluster using its default values. However, you can specify the cluster configurations as command line arguments to the `eksctl` command, or more conveniently, specify a YAML file containing your cluster configurations.
```